### DISCUSSION TITLE
Building a NextJS app using GitHub actions with the Vercel API gives Error: Blocked

### DISCUSSION THREAD

**Discussion by jpbnetley** [vercel/community#4778] [Help] [answered]
### Summary

Hi, any help would be appreciated.

I have a nextjs14 app that gets deployed via GitHub actions.
This uses the Vercel CLI to execute the deployment in the GitHub action.
npm is used to install the Vercel cli tool and to execute the commands as recommended by the [documentation](https://vercel.com/guides/how-can-i-use-github-actions-with-vercel).

This was working fine a few weeks ago, but started failing after merging in one of the dependency updates.

The GitHub action can be found here: https://github.com/jpbnetley/joke-of-the-day/blob/dev/.github/workflows/main.yml

When running the build locally with `pnpm run build`, it works as expected. 
Locally I am using node v21.1.0.
The GitHub action uses node 18

But when the GitHub action tries to execute the build, I get a ` Generating static pages (3/5) Error: Blocked` error.

### Example

https://github.com/jpbnetley/joke-of-the-day/actions/runs/6797392948/job/18479421016

### Steps to Reproduce

To reproduce, a merge request can be made to the dev branch, and the github action will try to deploy to the dev environment.

When the build process is executed, the ` Generating static pages (3/5) Error: Blocked` error is thrown.

Here is a snippet of the build action log.

The most recent build error is
```
> next build

Attention: Next.js now collects completely anonymous telemetry regarding usage.
This information is used to shape Next.js' roadmap and prioritize features.
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://nextjs.org/telemetry

   ▲ Next.js 14.0.1

   Creating an optimized production build ...
 ✓ Compiled successfully
   Linting and checking validity of types ...
   Collecting page data ...
   Generating static pages (0/5) ...

   Generating static pages (1/5) 

   Generating static pages (2/5) 

   Generating static pages (3/5) 
Error: Blocked
    at getJokesAsJson (/home/runner/work/joke-of-the-day/joke-of-the-day/.next/server/app/page.js:1:[64](https://github.com/jpbnetley/joke-of-the-day/actions/runs/6797392948/job/18479421016#step:13:65)91)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Home (/home/runner/work/joke-of-the-day/joke-of-the-day/.next/server/app/page.js:1:[65](https://github.com/jpbnetley/joke-of-the-day/actions/runs/6797392948/job/18479421016#step:13:66)55)
Error: Bad Gateway
    at getJokesAsJson (/home/runner/work/joke-of-the-day/joke-of-the-day/.next/server/app/page.js:1:6491)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Home (/home/runner/work/joke-of-the-day/joke-of-the-day/.next/server/app/page.js:1:6555)

Error occurred prerendering page "/". Read more: https://nextjs.org/docs/messages/prerender-error
Error: Blocked
    at getJokesAsJson (/home/runner/work/joke-of-the-day/joke-of-the-day/.next/server/app/page.js:1:64[91](https://github.com/jpbnetley/joke-of-the-day/actions/runs/6797392948/job/18479421016#step:13:92))
    at process.processTicksAndRejections (node:internal/process/task_queues:[95](https://github.com/jpbnetley/joke-of-the-day/actions/runs/6797392948/job/18479421016#step:13:96):5)
    at async Home (/home/runner/work/joke-of-the-day/joke-of-the-day/.next/server/app/page.js:1:6555)

 ✓ Generating static pages (5/5) 

> Export encountered errors on following paths:
	/page: /
 ELIFECYCLE  Command failed with exit code 1.
Error: Command "pnpm run build" exited with 1
Error: Process completed with exit code 1.
```

The main page where the error is thrown can be found here: https://github.com/jpbnetley/joke-of-the-day/blob/dev/src/app/page.tsx

**Comment by amyegan** [comment]
Please double check that your access token is set with the proper access and is not expired

**Reply by jpbnetley** [reply]
I had a look, and the token is set to never expire.
I can also see that the token was accessed when I reran the build now.

<img width="689" alt="image" src="https://github.com/vercel/community/assets/25173863/b6517743-4ba4-4562-ba63-e5b670ba93ca">

I logged in with my Github account, and see that the token has full access

If I recall, it looked like this when I created the token
<img width="254" alt="image" src="https://github.com/vercel/community/assets/25173863/578d05de-69ee-42e9-8cde-f77a3fae302e">

**Reply by amyegan** [reply]
Did you add `VERCEL_TOKEN`, `VERCEL_ORG_ID`, and `VERCEL_PROJECT_ID` as [secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets) in the GitHub workflow?

**Reply by jpbnetley** [reply]
I see I don't have VERCEL_ORG_ID or VERCEL_PROJECT_ID.

I have added them now under .
<img width="612" alt="image" src="https://github.com/vercel/community/assets/25173863/1e95cb09-7e61-449d-b2a7-e3370de2b83d">


But I still get the 
```
Generating static pages (3/5) 
Error: Blocked
```

**Reply by jpbnetley** [reply, marked as answer]
The cause of the issue was related too the reddit api returning "blocked" during the build process on Vercel when pre-rendering.
I fixed this by returning an empty array if the reddit api call failed.

https://github.com/jpbnetley/joke-of-the-day/commit/a651cd8b42d43f2d3e81feccfef06cac2649deb9#diff-014f5bfe21fd3276036a8ff5200b84d0bda2c09d89258742c02972e6c05f3a6e
